.circleci/config.yml

```diff
@@ -0,0 +1,191 @@
+          command: |
+            docker build -t ${DOCKHUB_ORGANISATION}/ticktrade:${CIRCLE_TAG} -t ${DOCKHUB_ORGANISATION}/ticktrade:<< parameters.target >>-latest .
+      - run:
+          name: Pushing Image to docker hub
+          command: |
+            echo $DOCKERHUB_PASSWORD | docker login -u $DOCKERHUB_USERNAME --password-stdin
+            docker push ${DOCKHUB_ORGANISATION}/ticktrade
+  k8s_deploy:
+    description: "Deploy to k8s cluster"
+    parameters:
+      k8s_svc_name:
+        type: string
+        default: "staging-ticktrade-binary-com"
+    steps:
+      - k8s/install-kubectl
+      - run:
+          name: Deploying to k8s cluster for service << parameters.k8s_svc_name >>
+          command: |
+            for SERVER_ID in {1..5}
+            do
+              KUBE_SERVER_REF="KUBE_SERVER_$SERVER_ID"
+              SERVICEACCOUNT_TOKEN_REF="SERVICEACCOUNT_TOKEN_$SERVER_ID"
+              CA_CRT_REF="CA_CRT_$SERVER_ID"
+              if [ ! -z "${!KUBE_SERVER_REF}" ]
+              then
+                echo "Deploying to cluster $SERVER_ID"
+                CA_CRT="${!CA_CRT_REF}"
+                KUBE_SERVER="${!KUBE_SERVER_REF}"
+                SERVICEACCOUNT_TOKEN="${!SERVICEACCOUNT_TOKEN_REF}"
+                echo $CA_CRT | base64 --decode > ca.crt
+                kubectl --server=${KUBE_SERVER} --certificate-authority=ca.crt --token=$SERVICEACCOUNT_TOKEN set image deployment/<< parameters.k8s_svc_name >> << parameters.k8s_svc_name >>=${DOCKHUB_ORGANISATION}/ticktrade:${CIRCLE_TAG}
+              fi
+            done
+
+  npm_test:
+    description: Run test
+    steps:
+      - run:
+          name: Run tests
+          command: |
+            for i in {1..3}; do npm test && break; done
+
+jobs:
+  test:
+    docker:
+      - image: circleci/node:11.15.0
+    steps:
+      - git_checkout_from_cache
+      - npm_install
+      - npm_test
+  release_staging:
+    docker:
+      - image: circleci/node:11.15.0
+    steps:
+      - git_checkout_from_cache
+      - npm_install
+      - build
+      - docker:
+          target: "staging"
+      - k8s_deploy:
+          k8s_svc_name: "staging-ticktrade-binary-com"
+  release_production:
+    docker:
+      - image: circleci/node:11.15.0
+    steps:
+      - git_checkout_from_cache
+      - npm_install
+      - build
+      - compress
+      - upload_artifact  # uploading the built code to s3 to create a backup of key services separate from Kubernetes deployment
+      - upload_checksum  # uploading compressed artifact checksum to cross match artifact fingerprint before actual deployment
+      - docker:
+          target: "production"
+      - k8s_deploy:
+          k8s_svc_name: "production-ticktrade-binary-com"
+
+workflows:
+  test:
+    jobs:
+      - test
+  release:
+    jobs:
+      - release_staging:
+          filters:
+            branches:
+              ignore: /.*/
+            tags:
+              only: /^staging.*/
```

Comment:
Can we add `context` here also. Not sure how `release_staging` will get those env variables if context is not provided.

Reply:
The `context` is added to `release_staging`.